src/zsclient.cpp

```diff
@@ -750,7 +758,7 @@ namespace zsync2 {
             // copy value for later use
             int okUrls = n;
 
-            if (!url) {
+            if (!url && userSpecifiedUrlisEmpty) {
```

Comment:
The check you added seems superfluent. You're setting `absoluteUrl`, so it should be sufficient to check whether `absoluteUrl` is set to some meaningful value, right?